R/mlRegressionNeuralNetwork.R

```diff
@@ -0,0 +1,326 @@
+  nodeNames <- strsplit(weightNames, split = "[.]to[.]")
+  from_node <- unlist(lapply(nodeNames, `[[`, 1))
+  to_node <- unlist(lapply(nodeNames, `[[`, 2))
+  
+  layerNames <- strsplit(to_node, split = "layhid")
+  to_layer <- as.numeric(unlist(lapply(layerNames, `[[`, 1)))
+  to_layer[which(is.na(to_layer))] <- max(to_layer, na.rm = TRUE) + 1
+  from_layer <- to_layer - 1
+  
+  hidden <- .getNeuralNetworkNodes(options)
+  
+  for (i in 1:length(hidden)) {
+    from_node <- gsub(pattern = paste0(i, "layhid"), from_node, replacement = "Hidden ")
+    to_node <- gsub(pattern = paste0(i, "layhid"), to_node, replacement = "Hidden ")
+  }
+  
+  from_layer[which(from_layer == 0)] <- "input"
+  from_layer[which(from_node == "Intercept")] <- NA
+  to_layer[which(to_layer == max(to_layer))] <- "output"
+  
+  table[["fromNode"]]  <- from_node
+  table[["fromLayer"]] <- from_layer
+  table[["separator"]] <- rep("\u2192", length(from_node))
+  table[["toNode"]]    <- to_node
+  table[["toLayer"]]   <- to_layer
+  table[["value"]]     <- weights
+  
+}
+
+.neuralNetworkGraph <- function(dataset, options, jaspResults, ready, purpose, position) {
+  
+  if (!is.null(jaspResults[["networkGraph"]]) || !options[["networkGraph"]]) return()
+  
+  plot <- createJaspPlot(title = gettext("Network Structure Plot"), height = 500, width = 500)
+  plot$position <- position
+  plot$dependOn(options = c("networkGraph", "plotEdgeLabels", "plotNetworkLegend", "scaleEqualSD", "target", "predictors", "seed", "seedBox", "holdoutData", "testDataManual",
+                            "threshold", "algorithm", "learningrate", "err.fct", "act.fct", "layers"))
+  jaspResults[["networkGraph"]] <- plot
+  
+  if(!ready)  return()
+  
+  result <- switch(purpose, 
+                   "classification" = jaspResults[["classificationResult"]]$object,
+                   "regression" = jaspResults[["regressionResult"]]$object)
+  model <- result[["model"]]
+  
+  weights <- model$result.matrix
+  index <- which(rownames(weights) == "steps")
+  weights <- weights[-(1:index),]
+  
+  weightNames <- names(weights)
+  nodeNames <- strsplit(weightNames, split = "[.]to[.]")
+  from_node <- unlist(lapply(nodeNames, `[[`, 1))
+  to_node <- unlist(lapply(nodeNames, `[[`, 2))
+  
+  hidden <- .getNeuralNetworkNodes(options)
+  
+  for (i in 1:length(hidden)) {
+    from_node <- gsub(pattern = paste0(i, "layhid"), from_node, replacement = paste0(i, "_Hidden "))
+    to_node <- gsub(pattern = paste0(i, "layhid"), to_node, replacement = paste0(i, "_Hidden "))
+  }
+  
+  allnames <- unique(c(from_node, to_node))
+  edgelist <- matrix(NA, nrow = length(allnames), ncol = length(allnames))
+  rownames(edgelist) <- allnames
+  colnames(edgelist) <- allnames
+  
+  for(i in 1:length(from_node)){
+    rowI <- which(rownames(edgelist) == from_node[i])
+    colI <- which(colnames(edgelist) == to_node[i])
+    edgelist[rowI, colI] <- weights[i]
+  }
+  
+  y.inc <- 1 / (length(hidden) + 1)
```

Comment:
These dot variables clash with S3 philosophy so it's better not to specify them in camelCase instead.